core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt

```diff
@@ -11,46 +11,54 @@ import ch.ergon.dope.validtype.StringType
 import ch.ergon.dope.validtype.ValidType
 
 sealed class Primitive<T : ValidType>(
-    private val dopeQuery: DopeQuery,
+    private val dopeQuery: () -> DopeQuery,
```

Comment:
since it is a function rename it to generateDopeQuery() or something

Reply:
This change is only needed due to the _ArrayPrimitive_ right?

Reply:
yes